lib/utils/getFileIgnorer.mjs

```diff
@@ -25,15 +23,16 @@ export default function getFileIgnorer({ ignorePath, ignorePattern, cwd }) {
 			? ignoreFilePath
 			: resolve(cwd, ignoreFilePath);
 
-		try {
-			const ignoreText = readFileSync(absoluteIgnoreFilePath, 'utf8');
+		if (!existsSync(absoluteIgnoreFilePath)) continue;
+
+		const ignoreText = readFileSync(absoluteIgnoreFilePath, {
+			// utf must remain lowercased to hit the fast path
+			// see nodejs/node#49888
+			encoding: 'utf8',
+			flag: 'rs',
```

Comment:
[ask] Just curious, what difference is between `rs` and unspecified for `readFileSync`? The [Node.js doc](https://nodejs.org/api/fs.html#file-system-flags) says:

> `'rs'`: Open file for reading in synchronous mode. An exception occurs if the file does not exist.

The default flag is `'r'`. Ref: https://nodejs.org/api/fs.html#fsreadfilesyncpath-options

Reply:
I just wanted to be specific about it.
In this case it might not make much difference with the default.
I shouldn't change this if not backed up by a benchmark.